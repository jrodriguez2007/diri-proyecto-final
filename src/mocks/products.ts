import { Product } from '../models/Product';

export const mockProducts: Product[] = [
  {
    id: 1,
    codigo: 'PROD001',
    descripcion: 'SAMSUNG ODYSSEY LS27CG552EUXEN 27" LED QHD 165HZ FREESYNC CURVO',
    idMarca: 1,
    marca: 'SAMSUNG',
    idModelo: 1,
    modelo: 'ODYSSEY',
    idFamilia: 3,
    familia: 'MONITORES LED',
    idUnidad: 8,
    unidad: 'UNIDAD ',
    activo: true,
    idUsuarioRegistro: "admin"
  },
  {
    id: 2,
    codigo: 'PROD002',
    descripcion: 'SAMSUNG 49" OLED - ODYSSEY OLED G9 G91SD S49DG910SU',
    idMarca: 1,
    marca: 'SAMSUNG',
    idModelo: 2,
    modelo: 'FREESYNC CURVO',
    idFamilia: 3,
    familia: 'MONITORES LED',
    idUnidad: 8,
    unidad: 'UNIDAD ',
    activo: true,
    idUsuarioRegistro: "admin"
  },
  {
    id: 3,
    codigo: 'PROD003',
    descripcion: "MONITOR LG ULTRAWIDE 34' CURVO, VA, WQHD, 100 HZ, 21:9",
    idMarca: 1,
    marca: 'SAMSUNG',
    idModelo: 2,
    modelo: 'FREESYNC CURVO',
    idFamilia: 4,
    familia: 'MONITORES LCD',
    idUnidad: 3,
    unidad: 'DOCENA',
    activo: true,
    idUsuarioRegistro: "admin"
  }
];
