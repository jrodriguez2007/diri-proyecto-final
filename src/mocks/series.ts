import { Serie } from "../models/Serie";

export const mockSeries: Serie[] = [
  {
    id: "E001",
    idDocument: "PE",
    idSerie: "E001",
    number: 1,
    currentNumber: 100,
    sizeText: 5,
    status: true    
  },
  {
    id: "S001",
    idDocument: "PS",
    idSerie: "S001",
    number: 1,
    currentNumber: 200,
    sizeText: 5,
    status: true     
  },  
];