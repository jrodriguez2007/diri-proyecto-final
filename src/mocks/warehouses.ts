import { Warehouse } from '../models/Warehouse';


export const mockWarehouses: Warehouse[] = [
  {
    id: 1,
    nombre: 'ALM. PRINCIPAL',
  },
];
