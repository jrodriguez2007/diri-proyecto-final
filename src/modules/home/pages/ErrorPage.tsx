import React from "react";
import { useTransform } from "@/hooks/useTransform";


export const ErrorPage: React.FC = () => {
  const { t } = useTransform();

  return (
    <>
      <h1>Error Page</h1>
    </>
  );
};
