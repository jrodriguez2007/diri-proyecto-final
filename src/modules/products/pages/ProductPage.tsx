import React, { useState, ChangeEvent, Suspense } from 'react';
import { Space } from 'antd';
import { FormattedMessage } from "react-intl";
import { ColumnType } from 'antd/es/table';
import { EditFilled, DeleteFilled } from '@ant-design/icons';

// Se importan las funciones de manejo (handlers) que se definieron en otro módulo (handlers.ts)
import { useGetAllProductsQuery } from '../store';
import { ContentLayout } from '@/modules/home/layout/ContentLayout';
import { useTransform } from '@/hooks/useTransform';
import { PaginationTemplate, TableTemplate } from '@/modules/templates/components';
import { HeaderButtonsTemplate } from '@/modules/templates/components/inputs';
// Se importan tanto las columnas principales como los handlers (lógica de negocio) de roles
import { useProductColumns, productHandlers, useProductSearch } from '../hooks';
import { Product } from '@/models/Product';
import LoggerService from '@/providers/LoggerService';

const FormComponent = React.lazy(() => import('../components/FormComponent'));
const ConfirmDeleteModal = React.lazy(() => import('../components/ConfirmDeleteComponent'));

export const ProductPage: React.FC = () => {
  const { t } = useTransform();

  // ---- Estados para la paginación ----
  const [pageSize, setPageSize] = useState<number>(10);
  // currentPage: número de la página actual.  
  const [currentPage, setCurrentPage] = useState(1);

  // ---- Estados para el formulario (crear/editar) ----
  const [isModalFormVisible, setIsModalFormVisible] = useState(false);
// formTitle: título que se muestra en el modal (puede cambiar entre "Crear" o "Editar").  
  const [formTitle, setFormTitle] = useState<string>(t('modules.form.create.title', {item: t('app.products.title.singular').toLocaleLowerCase()}));
  // formInitialValues: valores iniciales para precargar el formulario en modo edición.  
  const [formInitialValues, setFormInitialValues] = useState<any>(null);
  // isEditMode: indica si el formulario se está utilizando en modo edición (true) o creación (false).  
  const [isEditMode, setIsEditMode] = useState(false);

  // ---- Estados para el proceso de eliminación ----
  const [isModalDeleteVisible, setIsModalDeleteVisible] = useState(false);
  // selectedRecord: almacena el registro (Product) seleccionado para eliminar.  
  const [selectedRecord, setSelectedRecord] = useState<Product | null>(null);

  // ---- Estado para el texto de búsqueda ----
  const [searchText, setSearchText] = useState<string>('');  

  // ---- Consulta para obtener la lista de roles de autor ----
  const { data: prevAllProducts, error, isLoading } = useGetAllProductsQuery(
    { page: 1, limit: 50 },
    // { skip: searchText.trim() !== '' } // Ejecutar la consulta solo cuando searchText este vacío
  );
  const allProducts: Product[] = (prevAllProducts ?? []) as Product[];

  // Se usa el custom hook de búsqueda; si searchText está vacío, se skipea la consulta.
  const { data: searchResults } = useProductSearch(searchText, 0);

  // Si hay búsqueda, se muestran los resultados filtrados; de lo contrario se muestran todos los roles.
  // const rolesData: Product[] | undefined =
  // searchText.trim() !== '' ? (searchResults as Product[]) : (allRoles as Product[]);

 // Filtrado en cliente: por descripción (case-insensitive)
 const rolesData = allProducts.filter((p) =>
  p.descripcion.toLowerCase().includes(searchText.toLowerCase())
);
  
  // Se realiza la paginación local: se extrae el subconjunto de registros según la página y el tamaño actual.  
  const paginatedData = (rolesData  ?? []).slice((currentPage - 1) * pageSize, currentPage * pageSize);

  // ---- Extracción de funciones del handler ----
  const {
    handleCreate,
    handleEdit,
    handleFormSubmit,
    handleDelete,
    confirmDelete,
    handleUpdateState,
  } = productHandlers();


// Función para manejar el cambio de paginación.
// Recibe un objeto con currentPage y pageSize y actualiza los estados correspondientes.
const handlePaginationChange = ({ currentPage, pageSize }: { currentPage: number, pageSize: number }) => {
  setCurrentPage(currentPage);
  setPageSize(pageSize);
};  


  // ---- Handler para el cambio en el textbox de búsqueda ----
  const onSearchChange = (e: ChangeEvent<HTMLInputElement>) => {
    setSearchText(e.target.value);
    // Reiniciamos la paginación al buscar
    setCurrentPage(1);
  };  

  // ---- Función para el envío del formulario ----
  const onSubmit = (values: any) => {
    handleFormSubmit(values, isEditMode, formInitialValues)
      .then((response) => {
        const msg = isEditMode
          ? 'Producto actualizado: '
          : 'Producto creado: ';
        LoggerService.info(`${msg} ${JSON.stringify(response, null, 2)}`);          
      })
      .catch((error) => {
        const msg = isEditMode
          ? 'Error al actualizar producto: ' 
          : 'Error al crear producto: ';
        LoggerService.error(`${msg} ${JSON.stringify(error, null, 2)}`);          
      });
    // Se cierra el modal del formulario tras el envío.
    setIsModalFormVisible(false);
  };

  // ---- Tabla: Columnas ---
  const productColumns = useProductColumns();
  // ---- Tabla: Acciones ---
  const actionColumnTable: ColumnType<Product>[] = [
    {
      title: t('table.header.actions'),
      key: 'acciones',
      render: (_: unknown, record: Product) => (
        <Space size="middle">
          <EditFilled onClick={() => handleEdit( record, setFormInitialValues, setIsModalFormVisible, setFormTitle, setIsEditMode)}
          />
          <DeleteFilled onClick={() => handleDelete(record, setSelectedRecord, setIsModalDeleteVisible)}
          />
        </Space>
      ),
    },
  ];    

  return (

    <ContentLayout 
      title={<FormattedMessage id="app.products.title" />}
      headerExtras={
      <HeaderButtonsTemplate
        searchText={searchText}
        searchPlaceholder={t(
          'modules.form.placeholder.search',
          {
            item: t('app.products.title').toLocaleLowerCase()
          }
        )}
        onSearchChange={onSearchChange}
        onFilterClick={() => console.log('Botón filtro presionado')}
        onAddClick={() => handleCreate(setIsEditMode, setFormTitle, setFormInitialValues, setIsModalFormVisible)}
      />}
    >

      <>
          {/* Contenido: Inicio */}

          {/* Se muestran mensajes de carga o error en caso de que la consulta falle */}
          {isLoading && <p><FormattedMessage id="message.loading" /></p>}
          {error && <p><FormattedMessage id="error.data" /></p>}

          {/* Si se obtuvieron datos (lista no vacía), se renderizan la tabla y la paginación */}
          {rolesData  && rolesData .length > 0 && (
            <>
              <TableTemplate 
                data={paginatedData} 
                columns={[...productColumns, ...actionColumnTable]}    
              />

              <PaginationTemplate 
                total={rolesData.length}
                pageSizeOptions={[5,10]}
                defaultPageSize={pageSize} 
                onPageChange={handlePaginationChange}
              />
            </>
          )}

          {/* FormComponent es el componente modal que contiene el formulario para crear o editar.
          Se le pasan los estados de visibilidad, título, valores iniciales y los handlers onSubmit y onCancel */}
          <Suspense fallback={<div>Cargando formulario…</div>}>
            {isModalFormVisible && (
              <FormComponent 
                visible={isModalFormVisible}
                title={formTitle}
                initialValues={formInitialValues}
                onSubmit={onSubmit}
                onCancel={() => setIsModalFormVisible(false)}
              />
            )}
          </Suspense>

          {/* Modal Confirmación Eliminación */}
          <Suspense fallback={null}>
            {isModalDeleteVisible && selectedRecord && (
              <ConfirmDeleteModal
                visible={isModalDeleteVisible}
                record={selectedRecord}
                onConfirm={() => {
                  confirmDelete(selectedRecord, setIsModalDeleteVisible, setSelectedRecord);
                  LoggerService.info('Confirmación de eliminación del producto');
                }}
                onCancel={() => {
                  setIsModalDeleteVisible(false);
                  setSelectedRecord(null);
                }}
              />
            )}
          </Suspense>          

        {/* Contenido: Fin */}

      </>
          
    </ContentLayout>

  );
};
