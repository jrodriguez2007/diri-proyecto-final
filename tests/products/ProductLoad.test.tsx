// tests/products/ProductLoad.test.tsx

import '@testing-library/jest-dom'
import React from 'react'
import { render, screen, cleanup } from '@testing-library/react'
import { IntlProvider } from 'react-intl'
import esMessages from '../../src/lang/es.json' 
import { afterEach, describe, expect, it, vi } from 'vitest'

import { mockProducts } from '../../src/mocks/products';

// vi.mock('react-intl', async (importOriginal) => {
//   const actual = await importOriginal<typeof import('react-intl')>()

//   return {
//     ...actual,
//     useIntl: () => ({
//       formatMessage: ({ id }) => id,
//     }),
//     FormattedMessage: ({ id }: { id: string }) => <>{id}</>,
//   }
// })

vi.mock('../../src/hooks/useTransform', () => ({
  useTransform: () => ({ t: (id: string) => id })
}))

// 1) Mocks de los subcomponentes lazy:
vi.mock(
  '../../../src/modules/products/components/FormComponent',
  () => ({ default: () => <div data-testid="form-component-mock" /> })
)
vi.mock(
  '../../../src/modules/products/components/ConfirmDeleteComponent',
  () => ({ default: () => <div data-testid="confirm-delete-mock" /> })
)

// 2) Mock del store y hooks (igual que antes):
vi.mock('../../src/modules/products/store', () => ({
  useGetAllProductsQuery: vi.fn(() => ({
    data: mockProducts,
    isLoading: false,
    error: undefined,
  })),
}))
vi.mock('../../src/modules/products/hooks', () => ({
  useProductSearch: vi.fn(() => ({ data: [], isLoading: false, error: undefined })),
  productHandlers: vi.fn(() => ({
    handleCreate: vi.fn(),
    handleEdit: vi.fn(),
    handleFormSubmit: vi.fn(),
    handleDelete: vi.fn(),
    confirmDelete: vi.fn(),
    handleUpdateState: vi.fn(),
  })),
  useProductColumns: vi.fn(() => [
    { title: 'Código', dataIndex: 'codigo', key: 'codigo' },
    { title: 'Descripción', dataIndex: 'descripcion', key: 'descripcion' },
  ]),
}))

// 3) Importa el componente **ya** con los lazy components mockeados
import { ProductPage } from '../../src/modules/products/pages/ProductPage'

afterEach(() => {
  cleanup()
  vi.clearAllMocks()
})

describe('ProductPage — Renderizado inicial', () => {


  it('Debe de mostrarse correctamente con valores por defecto', () => {
    const { container } = render(
      <IntlProvider locale="es" messages={esMessages}>
        <ProductPage />
      </IntlProvider>
    )   
    expect( container ).toMatchSnapshot();
  });    

  it('muestra el título y la tabla con contenido', () => {

    const { container } = render(
      <IntlProvider locale="es" messages={esMessages}>
        <ProductPage />
      </IntlProvider>
    )    

    // A) El título internacionalizado "Productos"
    expect(screen.getByText(esMessages['app.products.title'])).toBeInTheDocument()
    expect(screen.getByText(esMessages['buttons.new'])).toBeInTheDocument()

    // B) La tabla aparece
    const table = screen.getByRole('table')
    expect(table).toBeInTheDocument()

    // C) Las cabeceras moquedas
    expect(screen.getByText(esMessages['app.products.header.code'])).toBeInTheDocument()
    expect(screen.getByText(esMessages['app.products.header.description'])).toBeInTheDocument()    

    // D) Sin filas
    const tbody = table.querySelector('tbody')
    expect(tbody!.children.length).toBe(mockProducts.length)
  })
})