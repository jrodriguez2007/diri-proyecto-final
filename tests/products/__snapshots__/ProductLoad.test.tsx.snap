// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductPage — Renderizado inicial > Debe de mostrarse correctamente con valores por defecto 1`] = `
<div>
  <div
    class="ant-layout css-dev-only-do-not-override-tjsggz"
    style="height: 100%; width: 100%; overflow: hidden;"
  >
    <header
      class="ant-layout-header css-dev-only-do-not-override-tjsggz"
      style="background: transparent; display: flex; justify-content: space-between; align-items: center; font-size: 24px; font-weight: bold; padding: 16px;"
    >
      <div>
        Productos
      </div>
      <div>
        <div
          class="ant-space css-dev-only-do-not-override-tjsggz ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
        >
          <div
            class="ant-space-item"
          >
            <span
              class="ant-input-affix-wrapper css-dev-only-do-not-override-tjsggz ant-input-outlined"
              style="width: 250px;"
            >
              <span
                class="ant-input-prefix"
              >
                <span
                  aria-label="search"
                  class="anticon anticon-search"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="search"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                    />
                  </svg>
                </span>
              </span>
              <input
                class="ant-input css-dev-only-do-not-override-tjsggz"
                placeholder="modules.form.placeholder.search"
                type="text"
                value=""
              />
              <span
                class="ant-input-suffix"
              >
                <button
                  class="ant-input-clear-icon ant-input-clear-icon-hidden"
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="close-circle"
                    class="anticon anticon-close-circle"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close-circle"
                      fill="currentColor"
                      fill-rule="evenodd"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
                      />
                    </svg>
                  </span>
                </button>
              </span>
            </span>
          </div>
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-tjsggz ant-btn-primary ant-btn-color-primary ant-btn-variant-solid"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <span
                  aria-label="plus"
                  class="anticon anticon-plus"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="plus"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                    />
                    <path
                      d="M192 474h672q8 0 8 8v60q0 8-8 8H160q-8 0-8-8v-60q0-8 8-8z"
                    />
                  </svg>
                </span>
              </span>
              Nuevo
            </button>
          </div>
        </div>
      </div>
    </header>
    <main
      class="ant-layout-content css-dev-only-do-not-override-tjsggz"
      style="display: flex; justify-content: center; align-items: flex-start; min-height: 100vh; padding: 20px; background: #fff; border-radius: 8px; box-shadow: 0 2px 8px rgba(0, 0, 0, 0.1); margin: 20px; overflow: hidden; box-sizing: border-box;"
    >
      <div
        style="width: 100%;"
      >
        <div
          class="ant-table-wrapper css-dev-only-do-not-override-tjsggz"
        >
          <div
            class="ant-spin-nested-loading css-dev-only-do-not-override-tjsggz"
          >
            <div
              class="ant-spin-container"
            >
              <div
                class="ant-table css-dev-only-do-not-override-tjsggz"
              >
                <div
                  class="ant-table-container"
                >
                  <div
                    class="ant-table-content"
                  >
                    <table
                      style="table-layout: auto;"
                    >
                      <colgroup />
                      <thead
                        class="ant-table-thead"
                      >
                        <tr>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            Código
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            Descripción
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            table.header.actions
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="ant-table-tbody"
                      >
                        <tr
                          class="ant-table-row ant-table-row-level-0 bg-gray-100"
                          data-row-key="1"
                        >
                          <td
                            class="ant-table-cell"
                          >
                            PROD001
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            SAMSUNG ODYSSEY LS27CG552EUXEN 27" LED QHD 165HZ FREESYNC CURVO
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="ant-space css-dev-only-do-not-override-tjsggz ant-space-horizontal ant-space-align-center ant-space-gap-row-middle ant-space-gap-col-middle"
                            >
                              <div
                                class="ant-space-item"
                              >
                                <span
                                  aria-label="edit"
                                  class="anticon anticon-edit"
                                  role="img"
                                  tabindex="-1"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="edit"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32zm-622.3-84c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9z"
                                    />
                                  </svg>
                                </span>
                              </div>
                              <div
                                class="ant-space-item"
                              >
                                <span
                                  aria-label="delete"
                                  class="anticon anticon-delete"
                                  role="img"
                                  tabindex="-1"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="delete"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-200 0H360v-72h304v72z"
                                    />
                                  </svg>
                                </span>
                              </div>
                            </div>
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0 bg-white"
                          data-row-key="2"
                        >
                          <td
                            class="ant-table-cell"
                          >
                            PROD002
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            SAMSUNG 49" OLED - ODYSSEY OLED G9 G91SD S49DG910SU
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="ant-space css-dev-only-do-not-override-tjsggz ant-space-horizontal ant-space-align-center ant-space-gap-row-middle ant-space-gap-col-middle"
                            >
                              <div
                                class="ant-space-item"
                              >
                                <span
                                  aria-label="edit"
                                  class="anticon anticon-edit"
                                  role="img"
                                  tabindex="-1"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="edit"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32zm-622.3-84c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9z"
                                    />
                                  </svg>
                                </span>
                              </div>
                              <div
                                class="ant-space-item"
                              >
                                <span
                                  aria-label="delete"
                                  class="anticon anticon-delete"
                                  role="img"
                                  tabindex="-1"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="delete"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-200 0H360v-72h304v72z"
                                    />
                                  </svg>
                                </span>
                              </div>
                            </div>
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0 bg-gray-100"
                          data-row-key="3"
                        >
                          <td
                            class="ant-table-cell"
                          >
                            PROD003
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            MONITOR LG ULTRAWIDE 34' CURVO, VA, WQHD, 100 HZ, 21:9
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="ant-space css-dev-only-do-not-override-tjsggz ant-space-horizontal ant-space-align-center ant-space-gap-row-middle ant-space-gap-col-middle"
                            >
                              <div
                                class="ant-space-item"
                              >
                                <span
                                  aria-label="edit"
                                  class="anticon anticon-edit"
                                  role="img"
                                  tabindex="-1"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="edit"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32zm-622.3-84c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9z"
                                    />
                                  </svg>
                                </span>
                              </div>
                              <div
                                class="ant-space-item"
                              >
                                <span
                                  aria-label="delete"
                                  class="anticon anticon-delete"
                                  role="img"
                                  tabindex="-1"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="delete"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-200 0H360v-72h304v72z"
                                    />
                                  </svg>
                                </span>
                              </div>
                            </div>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <ul
            class="ant-pagination ant-pagination-center css-dev-only-do-not-override-tjsggz"
          >
            <li
              aria-disabled="true"
              class="ant-pagination-prev ant-pagination-disabled"
              title="Previous Page"
            >
              <button
                class="ant-pagination-item-link"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  aria-label="left"
                  class="anticon anticon-left"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="left"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                    />
                  </svg>
                </span>
              </button>
            </li>
            <li
              class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
              tabindex="0"
              title="1"
            >
              <a
                rel="nofollow"
              >
                1
              </a>
            </li>
            <li
              aria-disabled="true"
              class="ant-pagination-next ant-pagination-disabled"
              title="Next Page"
            >
              <button
                class="ant-pagination-item-link"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  aria-label="right"
                  class="anticon anticon-right"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="right"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                    />
                  </svg>
                </span>
              </button>
            </li>
            <li
              class="ant-pagination-options"
            >
              <div
                aria-label="Page Size"
                class="ant-select ant-select-outlined ant-pagination-options-size-changer css-dev-only-do-not-override-tjsggz ant-select-single ant-select-show-arrow ant-select-show-search"
              >
                <div
                  class="ant-select-selector"
                >
                  <span
                    class="ant-select-selection-wrap"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-autocomplete="list"
                        aria-controls="rc_select_TEST_OR_SSR_list"
                        aria-expanded="false"
                        aria-haspopup="listbox"
                        aria-label="Page Size"
                        aria-owns="rc_select_TEST_OR_SSR_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="rc_select_TEST_OR_SSR"
                        role="combobox"
                        type="search"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-item"
                      title="10 components.pagination.locale"
                    >
                      10 components.pagination.locale
                    </span>
                  </span>
                </div>
                <span
                  aria-hidden="true"
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <span
                    aria-label="down"
                    class="anticon anticon-down ant-select-suffix"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </main>
  </div>
</div>
`;
